### ISSUE TITLE
mapStateToProps - cannot use default "or" assignment

### ISSUE THREAD

**Issue by murtyjones** [reduxjs/react-redux#978]
`React ^16.4.0 | React-Redux ^5.0.7 | Redux ^4.0.0`

The following mapStateToProps function:
```
const mapStateToProps = (state, ownProps) => {
  const { auth: { userId }, profiles: { profilesById } } = state
  const profile = profilesById[userId] || {}
  return {
    profile,
    userId
  }
}
```

Yields this error:
```
invariant.js:42 Uncaught Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
```

However, when `|| {}` is removed from line three, the error ceases.

Been using this pattern for a long time and never seen this behavior before.

**Comment by markerikson** [comment]
Looks like some kind of stack overflow issue, but hard to say what's going on without seeing a project that reproduces the issue.  However, note that the error message there is coming from React, not React-Redux.

**Comment by cellog** [comment]
I would be curious to see if there is any change if you take the {} and define it outside mSTP like so:

```js
const blank = {}
const mapStateToProps = (state, ownProps) => {
  const { auth: { userId }, profiles: { profilesById } } = state
  const profile = profilesById[userId] || blank
  return {
    profile,
    userId
  }
}
```

If it does, most likely you're triggering some kind of change on profile update, and it gets a new blank one every time the user profile is empty. But what Mark said: need more info to reproduce the issue

**Comment by murtyjones** [comment]
@cellog Your idea worked as you expected. Caused me to go back and reexamine some HOCS I had attached to this component, and I realized that I had one dispatching an action on `componentWillUpdate` without first checking if it needed to. I converted to `componentDidUpdate` and added in a needed check and that fixed the defaulting issue altogether.

**Before**
```
    componentWillUpdate() {
      this.props.closeTopBar()
    }
```

**After**
```
    componentDidUpdate() {
      if (this.props.isTopBarOpen) {
        this.props.closeTopBar()
      }
    }
```

Thanks for the help!
